SU2_CFD/src/solvers/CIncEulerSolver.cpp

```diff
@@ -2310,6 +2310,30 @@ void CIncEulerSolver::BC_Inlet(CGeometry *geometry, CSolver **solver_container,
         break;
     }
 
+    /* Loop over markers and check if the vertex is on another marker. If we have found 
+       that a vertex is on a viscous wall, we impose the no-slip condition and the pressure 
+       from the interior.
+    */
+
+   /*--- loop over markers in geometry ---*/
+    for (auto iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
+      /*--- check it is a viscous wall ---*/
+      if( config->GetViscous_Wall(iMarker)) {
+        /*--- loop over vertices of the wall and check if it matches with our vertex ---*/
+        unsigned long jPoint;
+        for (auto jVertex = 0; jVertex < geometry->nVertex[iMarker]; jVertex++) {
+          jPoint = geometry->vertex[iMarker][jVertex]->GetNode();
+          if (jPoint==iPoint){
+            /*--- set the velocity to zero (no-slip boundary on viscous walls) ---*/
+            for (iDim = 0; iDim < nDim; iDim++)
+              V_inlet[iDim+prim_idx.Velocity()] = 0.0;
```

Comment:
You need to copy from the interior instead of setting to zero (because it can be a moving wall)

Reply:
The no-slip condition will still be preserved if we copy velocity from the interior?

Reply:
I should because we are setting the interior values, by interior I mean the "node" (like you are doing with pressure) not the normal neighbor